### PR TITLE
Feature: Add SDXL Turbo model.

### PR DESCRIPTION
**Pull request by jiexiong2016**
This PR added an example for deploying sd-xl-turbo model with nos.

### CODE REVIEW

File: examples/stable-diffusion-XL-turbo/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+## Inference
+Here is an example how to run the inference:
+``` python
+from PIL import Image
+from nos.client import Client
+
+client = Client("[::]:50051")
+model = client.Module("sd-xl-turbo")
+
+frames: List[Image.Image] = model.text2image(prompts=["astronaut on the moon, hdr, 4k"], num_inference_steps=1)
```
Comment (spillai) [inline examples/stable-diffusion-XL-turbo/README.md:21]: Can you make this just `model(...)` and rename `text2image` method in the class to `__call__`? Makes it simpler for single method classes.

File: examples/stable-diffusion-XL-turbo/models/sdxl_turbo.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+from typing import List, Union
+
+from PIL import Image
+import torch
+from diffusers import AutoPipelineForText2Image
+
+class StableDiffusionXLTurboModel():
```
Comment (spillai) [inline examples/stable-diffusion-XL-turbo/models/sdxl_turbo.py:7]: remove `()`: `class StableDiffusionXLTurboModel:` is sufficient

File: examples/stable-diffusion-XL-turbo/tests/test_sdxl_turbo.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    model_id = "sd-xl-turbo"
+    models: List[str] = client.ListModels()
+    assert model_id in models
+
+    model = client.Module(model_id)
+    assert model is not None
+    assert model.GetModelInfo() is not None
+    print(f"Test [model={model_id}]")
+
+    response: List[Image.Image] = model.text2image(prompts=["astronaut on the moon, hdr, 4k"], num_inference_steps=1)
```
Comment (spillai) [inline examples/stable-diffusion-XL-turbo/tests/test_sdxl_turbo.py:22]: Use `model(...)`

**Review by spillai — APPROVED** [review]
:lgtm pending edits.

File: examples/stable-diffusion-XL-turbo/models/sdxl_turbo.py (hunk 2 above)
Comment (jiexiong2016) [reply examples/stable-diffusion-XL-turbo/models/sdxl_turbo.py]: Yes, changed to `__call__`  in class.

File: examples/stable-diffusion-XL-turbo/README.md (hunk 1 above)
Comment (jiexiong2016) [reply examples/stable-diffusion-XL-turbo/README.md]: Yes, changed to `__call__`  in class.

File: examples/stable-diffusion-XL-turbo/models/sdxl_turbo.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        # Only support gpu for video generation
+        assert torch.cuda.is_available()
+
+        self.torch_dtype = getattr(torch, dtype)
+        self.variant = "fp16" if self.torch_dtype == torch.float16 else None
+
+        self.pipe = AutoPipelineForText2Image.from_pretrained(model_name, torch_dtype=self.torch_dtype, variant="fp16")
+        self.pipe.to("cuda")
+
+    def text2image(
```
[the remark this answers is not in the document]
Comment (jiexiong2016) [reply examples/stable-diffusion-XL-turbo/models/sdxl_turbo.py]: Yes, changed to `__call__`  in class.

File: examples/stable-diffusion-XL-turbo/serve.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+      - transformers==4.35.2
+      - accelerate==0.23.0
+    env:
+      NOS_LOGGING_LEVEL: DEBUG
+
+models:
+  sd-xl-turbo:
+    model_cls: StableDiffusionXLTurboModel
+    model_path: models/sdxl_turbo.py
+    default_method: text2image
```
[the remark this answers is not in the document]
Comment (jiexiong2016) [reply examples/stable-diffusion-XL-turbo/serve.yaml]: Yes, changed to `__call__`  in class.

**Pull request merged**